### PR TITLE
Init spot with mat

### PR DESCRIPTION
**Pull request by iamchrissmith**
This adds a `Spot.setMat` function to the spec so that we can call it when initializing the collateral.  

Previously the `mat` was set to 1 with the `Spot.init` function which was then excluded from the Solidity output causing either a divide by 0 error or needing to hard code the mat into the solidity test harness.

I did not find an obvious way to remove the `mat` from the init function so I opted to set it to 0, but open to a better way of doing this.

### CODE REVIEW

File: spot.md [hunk 1]
```diff
@@ -112,13 +112,17 @@ Because data isn't explicitely initialized to 0 in KMCD, we need explicit initia
 ```k
     syntax SpotAuthStep ::= "init"     String
                           | "setPrice" String Wad
+                          | "setMat"   String Ray
```
Comment (kmbarry1) [inline spot.md]: Why not `"file" String String Ray` with the appropriate reversion behavior to mirror the contract implementation?

Comment (kmbarry1) [reply spot.md]: https://github.com/makerdao/dss/blob/master/src/spot.sol#L87

Comment (iamchrissmith) [reply spot.md]: good catch

**Comment by iamchrissmith** [comment]
@kmbarry1 sorry I needed to run an errand and stepped away before I could post this. 

I am getting some *weird* output with `make test-execution` after `make building` this:

```
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-flap-end.mcd.out tests/attacks/lucash-flap-end.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-flap-end.random.mcd.out tests/attacks/lucash-flap-end.random.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-flip-end.mcd.out tests/attacks/lucash-flip-end.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-flip-end.random.mcd.out tests/attacks/lucash-flip-end.random.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-pot-end.mcd.out tests/attacks/lucash-pot-end.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-pot-end.random.mcd.out tests/attacks/lucash-pot-end.random.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-pot.mcd.out tests/attacks/lucash-pot.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/attacks/lucash-pot.random.mcd.out tests/attacks/lucash-pot.random.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/fixes/debtBoundDSR.mcd.out tests/fixes/debtBoundDSR.mcd.expected
git --no-pager diff --no-index --ignore-all-space -R tests/fixes/potChiPieDai.mcd.out tests/fixes/potChiPieDai.mcd.expected
diff --git b/tests/attacks/lucash-pot-end.random.mcd.expected a/tests/attacks/lucash-pot-end.random.mcd.out
index f464c61..e848bc0 100644
diff --git b/tests/attacks/lucash-pot.mcd.expected a/tests/attacks/lucash-pot.mcd.out
index 97899c0..254c0c1 100644
diff --git b/tests/attacks/lucash-pot.random.mcd.expected a/tests/attacks/lucash-pot.random.mcd.out
index b0a35c8..c013cf3 100644
--- b/tests/attacks/lucash-pot.random.mcd.expected
--- b/tests/attacks/lucash-pot-end.random.mcd.expected
--- b/tests/attacks/lucash-pot.mcd.expected
diff --git b/tests/attacks/lucash-pot-end.mcd.expected a/tests/attacks/lucash-pot-end.mcd.out
+++ a/tests/attacks/lucash-pot.random.mcd.out
+++ a/tests/attacks/lucash-pot.mcd.out
+++ a/tests/attacks/lucash-pot-end.random.mcd.out
index f7e583d..8f81a94 100644
--- b/tests/attacks/lucash-pot-end.mcd.expected
+++ a/tests/attacks/lucash-pot-end.mcd.out
diff --git b/tests/attacks/lucash-flip-end.random.mcd.expected a/tests/attacks/lucash-flip-end.random.mcd.out
index 57267fd..a431b3c 100644
--- b/tests/attacks/lucash-flip-end.random.mcd.expected
+++ a/tests/attacks/lucash-flip-end.random.mcd.out
@@ -324,14 +324,14 @@
@@ -642,7 +642,7 @@
             "gold" |-> Ilk ( 0 , 1 , 3000000000 , 1000000000000 , 0 )
       Vow |-> 0 , 0 , Flap |-> 0
@@ -642,7 +642,7 @@
@@ -324,14 +324,14 @@
           </vat-ilks>
       GemJoin "MKR" |-> 0
           <vat-urns>
       Vow |-> 0 ) )
             "gold" |-> Ilk ( 0 , 1 , 3000000000 , 1000000000000 , 0 )
       Vow |-> 0 , 0 , Flap |-> 0
-            { "gold" , "Alice" } |-> Urn ( 0 , 0 )
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
           </vat-ilks>
-            { "gold" , "Bobby" } |-> Urn ( 0 , 0 )
       GemJoin "MKR" |-> 0
           <vat-urns>
@@ -642,7 +642,7 @@
-            { "gold" , "Alice" } |-> Urn ( 0 , 0 )
+            { "gold" , "Alice" } |-> Urn ( 2999999999 /Rat 300000000 , 0 )
       Vow |-> 0 ) )
-            { "gold" , "Bobby" } |-> Urn ( 0 , 0 )
+            { "gold" , "Bobby" } |-> Urn ( 2999999999 /Rat 300000000 , 0 )
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
             { "gold" , End } |-> Urn ( 0 , 0 )
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
       Vow |-> 0 , 0 , Flap |-> 0
       ListItem ( Measure ( 0 , End |-> 0
           </vat-urns>
+            { "gold" , "Alice" } |-> Urn ( 2999999999 /Rat 300000000 , 0 )
       GemJoin "MKR" |-> 0
       Flap |-> 0
           <vat-gem>
+            { "gold" , "Bobby" } |-> Urn ( 2999999999 /Rat 300000000 , 0 )
       Vow |-> 0 ) )
             { "gold" , "Alice" } |-> 0
       Pot |-> 0
             { "gold" , "Bobby" } |-> 0
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
             { "gold" , End } |-> Urn ( 0 , 0 )
-            { "gold" , End } |-> 20
           </vat-urns>
@@ -652,7 +652,8 @@
+            { "gold" , End } |-> 1 /Rat 150000000
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
             { "gold" , Flip "gold" } |-> 0
       Vow |-> 0 , 0 , Flap |-> 0
       ListItem ( Measure ( 0 , End |-> 0
           </vat-gem>
       Flap |-> 0
           <vat-gem>
       GemJoin "MKR" |-> 0
       Flap |-> 0
       Vow |-> 0 ) )
       Pot |-> 0
             { "gold" , "Alice" } |-> 0
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       Pot |-> 0
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
             { "gold" , "Bobby" } |-> 0
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
@@ -652,7 +652,8 @@
-            { "gold" , End } |-> 20
       ListItem ( Measure ( 0 , End |-> 0
       Vow |-> 0 , 0 , Flap |-> 0
       Flap |-> 0
+            { "gold" , End } |-> 1 /Rat 150000000
       GemJoin "MKR" |-> 0
       Pot |-> 0
           <vat-dai>
       Vow |-> 0 ) )
@@ -652,7 +652,8 @@
             { "gold" , Flip "gold" } |-> 0
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
           </vat-gem>
       Vow |-> 0 , 0 , Flap |-> 0
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
@@ -642,7 +642,7 @@
           <vat-dai>
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
       GemJoin "MKR" |-> 0
       Vow |-> 0 , 0 , Flap |-> 0
       ListItem ( Measure ( 0 , End |-> 0
       Vow |-> 0 ) )
       GemJoin "MKR" |-> 0
       Flap |-> 0
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       Vow |-> 0 ) )
       Pot |-> 0
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
@@ -642,7 +642,7 @@
       ListItem ( Measure ( 0 , End |-> 0
       Flap |-> 0
       Vow |-> 0 , 0 , Flap |-> 0
       Flap |-> 0
       Pot |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )
@@ -652,7 +652,8 @@
       Pot |-> 0
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
       Vow |-> 0 , 0 , Flap |-> 0
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       GemJoin "MKR" |-> 0
       ListItem ( Measure ( 0 , End |-> 0
       Vow |-> 0 ) )
       Flap |-> 0
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       Pot |-> 0
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
@@ -652,7 +652,8 @@
       Flap |-> 0
       Vow |-> 0 , 0 , Flap |-> 0
       Pot |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
       Flap |-> 0
       Pot |-> 0
@@ -1130,7 +1131,7 @@
@@ -1127,7 +1128,7 @@
       GemJoin "MKR" |-> 0
       Flap |-> 0
       Vow |-> 0 ) )
       GemJoin "MKR" |-> 0
       ListItem ( GenStep ( b"d" , transact ANYONE End . skim "gold" "Alice" ) )
       Vow |-> 0 ) )
-      ListItem ( LogNote ( ANYONE , Vat . grab "gold" "Alice" End Vow -10 -10 ) )
-      ListItem ( LogNote ( ADMIN , Vat . grab "gold" "Alice" End Vow -10 -10 ) )
+      ListItem ( LogNote ( ANYONE , Vat . grab "gold" "Alice" End Vow -1 /Rat 300000000 -10 ) )
+      ListItem ( LogNote ( ADMIN , Vat . grab "gold" "Alice" End Vow -1 /Rat 300000000 -10 ) )
       ListItem ( LogNote ( ANYONE , End . skim "gold" "Alice" ) )
       ListItem ( LogNote ( ADMIN , End . skim "gold" "Alice" ) )
make:        ListItem ( Measure ( 20 , "Alice" |-> 10
       ListItem ( Measure ( 20 , "Alice" |-> 10
       "Bobby" |-> 10
       "Bobby" |-> 10
*** [tests/attacks/lucash-pot.mcd.run] Error 1@@ -1144,7 +1145,7 @@
@@ -1148,7 +1149,7 @@
       Flap |-> 0
       GemJoin "MKR" |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )

       Vow |-> 0 ) )
       ListItem ( GenStep ( b"d" , transact ANYONE End . skim "gold" "Bobby" ) )
-      ListItem ( LogNote ( ADMIN , Vat . grab "gold" "Bobby" End Vow -10 -10 ) )
-      ListItem ( LogNote ( ANYONE , Vat . grab "gold" "Bobby" End Vow -10 -10 ) )
+      ListItem ( LogNote ( ADMIN , Vat . grab "gold" "Bobby" End Vow -1 /Rat 300000000 -10 ) )
+      ListItem ( LogNote ( ANYONE , Vat . grab "gold" "Bobby" End Vow -1 /Rat 300000000 -10 ) )
       ListItem ( LogNote ( ADMIN , End . skim "gold" "Bobby" ) )
make:        ListItem ( LogNote ( ANYONE , End . skim "gold" "Bobby" ) )
*** Waiting for unfinished jobs....       ListItem ( Measure ( 20 , "Alice" |-> 10

       ListItem ( Measure ( 20 , "Alice" |-> 10
       "Bobby" |-> 10
       "Bobby" |-> 10
make: *** [tests/attacks/lucash-pot.random.mcd.run] Error 1
make: *** [tests/attacks/lucash-flip-end.random.mcd.run] Error 1
make: *** [tests/attacks/lucash-pot-end.random.mcd.run] Error 1
make: *** [tests/attacks/lucash-pot-end.mcd.run] Error 1
diff --git b/tests/fixes/potChiPieDai.mcd.expected a/tests/fixes/potChiPieDai.mcd.out
index 06836b3..9965338 100644
--- b/tests/fixes/potChiPieDai.mcd.expected
+++ a/tests/fixes/potChiPieDai.mcd.out
@@ -642,7 +642,7 @@
       Vow |-> 0 , 0 , Flap |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
       Flap |-> 0
       Pot |-> 0
@@ -652,7 +652,8 @@
       Vow |-> 0 , 0 , Flap |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
       Flap |-> 0
       Pot |-> 0
diff --git b/tests/fixes/debtBoundDSR.mcd.expected a/tests/fixes/debtBoundDSR.mcd.out
index 6541fb7..0978d91 100644
--- b/tests/fixes/debtBoundDSR.mcd.expected
+++ a/tests/fixes/debtBoundDSR.mcd.out
@@ -297,6 +297,7 @@
                 SetItem ( End )
                 SetItem ( GemJoin "gold" )
                 SetItem ( Pot )
+                SetItem ( Spot )
               </vat-wards>
               <vat-can>
                 "Alice" |-> SetItem ( End )
@@ -657,7 +658,7 @@
             .Set
           </spot-wards>
           <spot-ilks>
-            "gold" |-> SpotIlk ( 1 , 1 )
make: +            "gold" |-> SpotIlk ( 3000000000 , 1 )
*** [tests/fixes/potChiPieDai.mcd.run] Error 1           </spot-ilks>

           <spot-par>
             1
@@ -1007,7 +1008,7 @@
       Vow |-> 0 , 0 , Flap |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
       Flap |-> 0
       Pot |-> 0
@@ -1017,7 +1018,8 @@
       Vow |-> 0 , 0 , Flap |-> 0
       GemJoin "MKR" |-> 0
       Vow |-> 0 ) )
-      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
+      ListItem ( LogNote ( ANYONE , Vat . file spot "gold" 3000000000 ) )
+      ListItem ( Poke ( "gold" , 3000000000 , 3000000000 ) )
       ListItem ( Measure ( 0 , End |-> 0
       Flap |-> 0
       Pot |-> 0
make: *** [tests/fixes/debtBoundDSR.mcd.run] Error 1
```

When I compare the `.out` and `.expected` files, they seem the same to me.  Do you see anything obviously wrong here? Thanks!

**Comment by iamchrissmith** [comment]
@kmbarry1 I figured out my output updates and this seems to be working now. can you take a look as sanity check the changes to the math in the results now that we are using the actual spot price throughout?

**Comment by iamchrissmith** [comment]
@gbalabasquer here is the solidity output for this version of the random tester:
```
### Solidity
#### ID:0
------------

    // Test Run
    admin.vat_rely(address(goldJoin));
    admin.spotter_setPrice("gold", 3000000000);
    admin.spotter_file("mat", "gold", 1);
    admin.spotter_file("par", 1);
    admin.goldFlip_rely(address(end));
    admin.vat_file("Line", 1000000000000);
    admin.vat_file("line", "gold", 1000000000000);
    admin.vow_file("bump", 1000000000);
    admin.vow_file("hump", 0);
    admin.Gem_gold_mint(address(alice), 20);
    admin.Gem_gold_mint(address(bobby), 20);
    alice.vat_hope(address(pot));
    alice.vat_hope(address(goldFlip));
    alice.vat_hope(address(end));
    bobby.vat_hope(address(pot));
    bobby.vat_hope(address(goldFlip));
    bobby.vat_hope(address(end));
    alice.goldJoin_join(address(alice), 10);
    bobby.goldJoin_join(address(bobby), 10);
    alice.vat_move(address(alice), address(alice), 0);
    bobby.goldJoin_join(address(bobby), 65_/Rat_64);
    bobby.vat_hope(address(alice));
    admin.pot_file("dsr", 371_/Rat_320);
    hevm.warp(2);
    hevm.warp(2);
    hevm.warp(2);

    // Assertions
    assertTrue( Gems.cell() == GemCellMapItem( <gem-id>
  "gold"
</gem-id> , <gem>
  <gem-id>
    "gold"
  </gem-id>
  <gem-wards>
    .Set
  </gem-wards>
  <gem-balances>
    "bobby" |-> 575 /Rat 64
    "alice" |-> 10
    GemJoin "gold" |-> 1345 /Rat 64
  </gem-balances>
</gem> ) );
    assertTrue( pot.dsr() == 371 /Rat 320 );
    assertTrue( vat.can() == "bobby" |-> SetItem( Flip "gold" )
SetItem( "alice" )
SetItem( pot )
SetItem( end )
"alice" |-> SetItem( Flip "gold" )
SetItem( pot )
SetItem( end ) );
    assertTrue( vat.gem() == { "gold" , end } |-> 0
{ "gold" , Flip "gold" } |-> 0
{ "gold" , "bobby" } |-> 705 /Rat 64
{ "gold" , "alice" } |-> 10 );

------------
### /Solidity
```

**Pull request closed without merging**